functionary/ui/views/environments.py

```diff
@@ -1,28 +1,42 @@
+from django.contrib.auth.mixins import UserPassesTestMixin
 from django.shortcuts import redirect
 from django.views.decorators.http import require_POST
 from django.views.generic.detail import DetailView
 from django.views.generic.list import ListView
 
-from core.models import Environment, Package
+from core.auth import Permission
+from core.models import Environment, EnvironmentUserRole, Package
 
 
 class EnvironmentListView(ListView):
     model = Environment
 
     def get_queryset(self):
         """Sorts based on team name, then env name."""
-
-        return super().get_queryset().order_by("team__name", "name")
-
-
-class EnvironmentDetailView(DetailView):
+        if self.request.user.is_superuser:
+            return super().get_queryset().order_by("team__name", "name")
+        else:
+            environments = EnvironmentUserRole.objects.filter(
```

Comment:
This is the other spot I noticed missing the inheritance of roles assigned at the Team level.  I think I had a suggestion at some point that maybe having a helper on the User model would make sense, such that this could basically just be:

```python
return self.request.user.environments.all()
```

or some such thing.  Neither part (the Team inheritance or the helper) is mandatory for this PR, but at a minimum add a TODO for us to come back and do that.

Reply:
fixed